tests/iostability/test_iostability_degraded_path.py

```diff
@@ -0,0 +1,189 @@
+                                       obj_name_pref="object-", obj_size=size,
+                                       skip_cleanup=skip_cleanup, duration=None,
+                                       log_file_prefix=log_file_prefix, end_point=S3_CFG["s3_url"],
+                                       validate_certs=S3_CFG["validate_certs"])
+                self.log.info("Loop: %s Workload: %s objects of %s with %s parallel clients.",
+                              loop, samples, size, clients)
+                self.log.info("Log Path %s", resp[1])
+                assert not s3bench.check_log_file_error(resp[1]), \
+                    f"S3bench workload failed in loop {loop}. Please read log file {resp[1]}"
+                system_utils.remove_file(resp[1])
+                if skip_cleanup:
+                    self.log.info("Delete Created Objects")
+                    resp = self.s3t_obj.object_list(bucket_name=bucket_name)
+                    obj_list = resp[1]
+                    while len(obj_list):
+                        if len(obj_list) > 1000:
+                            self.s3t_obj.delete_multiple_objects(bucket_name, obj_list=obj_list[0:1000])
+                            obj_list = obj_list[1000:]
+                        else:
+                            self.s3t_obj.delete_multiple_objects(bucket_name=bucket_name, obj_list=obj_list)
+                            obj_list = []
+                    self.log.info("Objects deletion completed")
+            loop += 1
+
+    @pytest.mark.lc
+    @pytest.mark.io_stability
+    @pytest.mark.tags("TEST-40172")
+    def test_object_crud_single_pod_failure(self):
+        """Perform Object CRUD operations in degraded mode in loop using S3bench for 7 days"""
+        self.log.info("Step 1: Create 50 buckets in healthy mode ")
+        bucket_creation_healthy_mode = self.test_cfg['test_40172']['bucket_creation_healthy_mode']
+        bucket_list = None
+        if bucket_creation_healthy_mode:
+            resp = self.s3t_obj.create_multiple_buckets(50, 'test-40172')
+            bucket_list = resp[1]
+            self.log.info("Step 1: Bucket created in healthy mode ")
+        else:
+            self.log.info("Step 1: Skipped bucket creation in healthy mode ")
+
+        self.log.info("Step 2: Shutdown the data pod safely by making replicas=0")
+        self.log.info("Get pod name to be deleted")
+        pod_list = self.master_node_list[0].get_all_pods(pod_prefix=POD_NAME_PREFIX)
+        pod_name = random.sample(pod_list, 1)[0]
+        hostname = self.master_node_list[0].get_pod_hostname(pod_name=pod_name)
+
+        self.log.info("Deleting pod %s", pod_name)
+        resp = self.master_node_list[0].create_pod_replicas(num_replica=0, pod_name=pod_name)
+        assert_utils.assert_false(resp[0], f"Failed to delete pod {pod_name} by making replicas=0")
+        self.log.info("Step 2: Successfully shutdown/deleted pod %s by making replicas=0", pod_name)
```

Comment:
are we not planning to restore pod after the test is done?

Reply:
No, not as of now. The setup might be required for further analysis. hence not restoring/ redeploying as part of test.